### ISSUE TITLE
Best BT dongle for ling range

### ISSUE THREAD

**Issue by CheapB** [50ButtonsEach/fliclib-linux-hci#12]
I am using Flic to control my Sonos and it is working, however I cannot get the needed range with the bluetooth dongle I am using. Are you aware of dongles with better range than others? or if it is possible to adjust the signal strength via sw.

**Comment by pixelecho** [comment]
can you publish your sonos script?

**Comment by CheapB** [comment]
Yes, but it is likely not going to help you as all the fun happens on to the receiving end of the webservice, but here you are
`#!/usr/bin/env python3

# Test Client application.
#
# This program attempts to connect to all previously verified Flic buttons by this server.
# Once connected, it prints Down and Up when a button is pressed or released.
# It also monitors when new buttons are verified and connects to them as well. For example, run this program and at the same time the scan_wizard.py program.


import fliclib, requests

client = fliclib.FlicClient("localhost")


def sendsonos(_channel,_clicktype) :
    cmd = []
    if str(_clicktype) == "ClickType.ButtonSingleClick" :
        cmd.append("PlayPause")
    elif str(_clicktype) == "ClickType.ButtonDoubleClick" :
        cmd.append("Next")
    else :
        cmd.append("Toggle")
    r = requests.post("http://10.0.1.52:30052/COMMAND", json={ "Device" : "Sonos", "Room" : "Kitchen", "Command" : cmd })
    print (r)
    

def got_button(bd_addr):
	cc = fliclib.ButtonConnectionChannel(bd_addr)
	cc.on_button_single_or_double_click_or_hold = lambda channel, click_type, was_queued, time_diff: sendsonos(channel,click_type)
	cc.on_connection_status_changed = \
		lambda channel, connection_status, disconnect_reason: \
			print(channel.bd_addr + " " + str(connection_status) + (" " + str(disconnect_reason) if connection_status == fliclib.ConnectionStatus.Disconnected else ""))
	client.add_connection_channel(cc)
	

def got_info(items):
	print(items)
	for bd_addr in items["bd_addr_of_verified_buttons"]:
		got_button(bd_addr)

client.get_info(got_info)

client.on_new_verified_button = got_button

client.handle_events()
`

**Comment by devbobo** [comment]
What dongle are you currently using ? And range are you getting ? What range do you need ?

**Comment by CheapB** [comment]
I am using a generic CSR4.0 on an RPI2. 15ft is hit or miss. sometimes it works and some times it doesn't. I like to put a flic in every room but if I have to use 5-6 RPIs to run them due to distance it makes more sense to have WIFI buttons. I would need at least  25ft

**Comment by devbobo** [comment]
I'm using a BT dongle from Pluggable. It's attached to my RPi2 in a downstairs cupboard, with my button probably around 25ft away.

http://plugable.com/products/usb-bt4le

**Comment by UglyBob79** [comment]
I'm using the CSR 4.0 too, and I would say I get maximum 4 meters of range, so fairly useless even in my small apartment :( So I would be happy too if there are other dongles that extend the range at least up to 15-20 meters...

**Comment by UglyBob79** [comment]
Took a chance and bought an Asus dongle (USB-BT400) and can confirm that it sucks even more than the CSR with these buttons. Maybe 2 meters of range.

**Comment by Emill** [comment]
We bought this adapter: http://www.lawicel-shop.se/prod/Bluetooth-20EDR-USB-adapter_990068/Sena-Technologies_47457/SWE/SEK and we can confirm it has much better range than the cheaper small dongles such as Asus USB-BT400/Plugable. Don't know why the webshop says it's Bluetooth 2.0 because it actually has Bluetooth 4.0.

**Comment by UglyBob79** [comment]
I just got mine and I'm having troubles with it in Linux. Are you using Windows or have you installed some driver in Linux? Got this from flicd when it crashes:

Warning: Bluetooth controller does not support pending connection and scanning at the same time. Will emulate this as good as possible.
Initialization of Bluetooth controller done!
Accepted new client
Error: Bluetooth controller sent a Hardware Error event with code 0x58, resetting... If it doesn't work, replug your controller.
writev: File descriptor in bad state

**Comment by Emill** [comment]
Lol. No we used this Linux SDK directly.
Can you post the output of "sudo btmon" (plaintext so github does not mess up the formatting)?

**Comment by Emill** [comment]
@UglyBob79 Have you got this working yet or is it still giving you the same error?

**Comment by CheapB** [comment]
What kind of range are you getting with the plugable and parani dongles? I am now getting 60ft with the built-in bluetooth adaptor on C.H.I.P

**Comment by UglyBob79** [comment]
I actually gave up for now, I think I remember that even when not getting that error, the range was still a joke. Maybe twice as much as the other dongles, but still not even full range in my small 2 room apartment. So I don't know, I might give it a try again in the future, but right now I'm working with other stuff.
